### PR TITLE
Migrating to pyproject.toml

### PR DESCRIPTION
**Pull request by castelao**
@zqtzt , if you want, I can guide you through re-organizing your package. We would do it in small steps. This one would be about moving to `pryproject.toml`, which is a more modern way of doing it.

### CODE REVIEW

File: DC_OCEAN.egg-info/PKG-INFO
Comment (castelao) [inline DC_OCEAN.egg-info/PKG-INFO:1]: @zqtzt , The `.egg` directory shouldn't go into the repository.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+authors = [
+  {name = "Zhetao Tan", email = "tanzhetao@mail.iap.ac.cn"},
+  {name = "Xinyi Song", email = "songxinyi231@mails.ucas.ac.cn"},
+  {name = "Lijing Cheng"},
+]
+maintainers = [
+  {name = "Zhetao Tan", email = "tanzhetao@mail.iap.ac.cn"},
+]
+keywords = ["oceanography", "duplicate", "quality control"]
+requires-python = ">= 3.7"
```
Comment (castelao) [inline pyproject.toml:22]: @zqtzt , consider moving this to `3.10`, or if you have a good reason for not, at least `3.8`.

File: DC_OCEAN.egg-info/PKG-INFO
Comment (zqtzt) [reply DC_OCEAN.egg-info/PKG-INFO:1]: Got it. I will delete it

File: pyproject.toml (hunk 1 above)
Comment (zqtzt) [reply pyproject.toml:22]: Yes, we can move it to '>= 3.8'

**Review by zqtzt — APPROVED** [review]

**Review by zqtzt — APPROVED** [review]

**Comment by castelao** [comment]
@zqtzt , I think this is enough for this PR. I'll add more in another PR. If you agree with the suggestions from this PR, I suggest 'squash n merge' instead of just merging; otherwise, it will be too many small commits.

Take a look at:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges

Let me know if you need help.

**Comment by zqtzt** [comment]
Thanks Gui! I merged all the commits. But it seems missed the file 'DC_OCEAN/_version.py'. Could you check it?

**Comment by castelao** [comment]
That file is created whenever you build it, including when you install with pip. Try `pip install -e .`

**Pull request merged**